torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -281,6 +307,48 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                     node.replace_all_uses_with(guard_scalar(val))
                     graph.erase_node(node)
 
+    # There are sometimes when we actually didn't set any replacements but still
+    # have specializations. For these cases we scan through placeholders and specialize
+    # using node meta val. Here's an example graph where you have no replacements but
+    # still want to specialize and get rid of zf0 before inductor:
+    #
+    # def forward(self, L_tensor_: "f16[2, 3, 4][12, 4, 1]cuda:0", L_scale_factor_: "f64[][]cpu"):
+    #     l_tensor_ = L_tensor_
+    #     l_scale_factor_ = L_scale_factor_
+    #     item: "Sym(zf0)" = l_scale_factor_.item();  l_scale_factor_ = None
+    #     interpolate: "f16[2, 3, TruncToInt(4.0*zf0)][3*TruncToInt(4.0*zf0), TruncToInt(4.0*zf0), 1]cuda:0" = torch.nn.functional.interpolate(l_tensor_, size = None, scale_factor = item, mode = 'nearest-exact', align_corners = None, recompute_scale_factor = True);  l_tensor_ = item = None # noqa: B950
+    #     return (interpolate,)
+    for i, node in enumerate(graph.nodes):
```

Comment:
I have an experimental improvement in https://github.com/pytorch/pytorch/pull/140829

I'll wait for CI to get green before squashing but the high level idea is to instead maintain a whitelist of symfloats we managed to tensorify away and then specialize anything not in the whitelist. Makes a lot of this logic much cleaner.